src/envoy/http/service_control/filter_test.cc

```diff
@@ -50,13 +51,20 @@ class ServiceControlFilterTest : public ::testing::Test {
                                                      mock_handler_factory_);
     filter_->setDecoderFilterCallbacks(mock_decoder_callbacks_);
 
+    mock_handler_ = new testing::NiceMock<MockServiceControlHandler>();
+    mock_handler_ptr_.reset(mock_handler_);
+    ON_CALL(mock_handler_factory_, createHandler(_, _))
+        .WillByDefault(Return(ByMove(std::move(mock_handler_ptr_))));
+
     mock_span_ = std::make_unique<Envoy::Tracing::MockSpan>();
   }
 
   std::unique_ptr<ServiceControlFilter> filter_;
   testing::NiceMock<MockStreamDecoderFilterCallbacks> mock_decoder_callbacks_;
   testing::NiceMock<MockFactoryContext> mock_factory_context_;
   testing::NiceMock<MockServiceControlHandlerFactory> mock_handler_factory_;
+  testing::NiceMock<MockServiceControlHandler>* mock_handler_;
```

Comment:
mock_handler_ptr_ is its unique_ptr. But when createHandler() is called,  its pointer will be moved.  It will become nullptr.    There are some test cases that we need to call EXPECT_CALL with the mock after createHanler() is called.  That is why we need to save the raw pointer in this member variable so we can use it any time regardless before of after createHandler().